internal/command/deploy/machines.go

```diff
@@ -553,3 +566,27 @@ func determineAppConfigForMachines(ctx context.Context, envFromFlags []string, p
 
 	return appConfig, nil
 }
+
+func (md *machineDeployment) ProcessNames() (names []string) {
+	names = md.appConfig.ProcessNames()
+	if len(md.processGroups) > 0 {
+		names = lo.Filter(names, func(name string, _ int) bool {
+			return md.processGroups[name] != nil
+		})
+	}
+	return
+}
+
+// Validate validates app config for process groups in the deployment.
```

Comment:
This flattening+validation is something that can happen as part of appConfig.Validate() call. 

It is that or bring the appConfig.GroupConfig() functionality in here

Reply:
Thanks for the feedback, I took another pass at this in 31a8581a04d9209dbcb5536247f6c9837fc24c51 which moves everything into an `appConfig.ValidateGroups()` call.